java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture10/tasks/_1/FileRedactor.java

```diff
@@ -0,0 +1,54 @@
+package com.zoolatech.lecture10.tasks._1;
+
+import java.io.*;
+import java.util.Date;
+
+final class FileRedactor {
+
+    public static void showFiles(File file) throws NoFileException {
+        File[] files = file.listFiles();
+        int counter = 0;
+        for (File value : files) {
+            if (value.isFile()) {
+                System.out.println("File number " + ++counter + " : " + value.getName());
+            } else if (value.isDirectory()) {
+                System.out.println("Directory: " + value.getName());
+            } else {
+                throw new NoFileException();
+            }
+        }
+    }
+
+    public static void generateFiles(String string, File file) throws IOException, NoFileException {
+        switch (string) {
```

Comment:
Let's refactor this switch to be more generic. Current implementation depends on the file names. You can check if listFiles contains name provided by user and if not just write error.